### PR TITLE
user on welcome and profile pages

### PR DESCRIPTION
**Pull request by jackconwayiv**
Basic implementation of authenticated user data on welcome and profile pages. Still not 100% functional (regarding Chakra-UI Avatar referrerPolicy prop).

### CODE REVIEW

File: frontend/src/components/UserProfile.tsx [hunk 1]
```diff
@@ -1,45 +1,52 @@
...
-  Text,
-  UnorderedList,
-} from "@chakra-ui/react";
+import { Avatar, Button, Center, Flex, Text } from "@chakra-ui/react";
+import { useAsync } from "react-async-hook";
 import { Link } from "react-router-dom";
+import { getUserDetails } from "../services/songs";
 
 export default function UserProfile() {
+  const asyncUser = useAsync(async () => getUserDetails(), []);
```
Comment (samdatkins) [inline frontend/src/components/UserProfile.tsx:7]: This isn't important, but thought you might want to know: useAsync expects an async function as it's first param, which means you can just pass it `getUserDetails` for that.

File: frontend/src/components/UserProfile.tsx [hunk 2]
```diff
@@ -1,45 +1,52 @@
...
-          </Text>
-          <UnorderedList mb="2rem" alignItems="center">
-            <ListItem>Toggle Night Mode</ListItem>
-            <ListItem>Toggle Chord Display</ListItem>
-            <ListItem>Set 1, 2, or 3 Columns</ListItem>
-            <ListItem>Your Requested Songs</ListItem>
-            <ListItem>Your Songbooks</ListItem>
-          </UnorderedList>
-          <Flex justifyContent="center">
+      {user ? (
```
Comment (samdatkins) [inline frontend/src/components/UserProfile.tsx:15]: Meant to mention this while we were pairing but I forgot. We should probably use either a Skeleton or Spinner here. Spinner is easier, Skeleton probably gives a slightly better UX (but I don't remember what this part of the page looks like, so I don't know how easy it'd be to use here).

File: frontend/src/components/UserProfile.tsx [hunk 3]
```diff
@@ -1,45 +1,52 @@
...
+            <Avatar
+              referrerPolicy="no-referrer"
+              size="xl"
+              name={`${user?.first_name} ${user?.last_name}`}
+              src={user?.social.picture}
+            />
+          </Center>
+          <Center>{user?.email}</Center>
+          <Center>Member since {joinedDate.toDateString()}</Center>
+          <Center>
```
Comment (samdatkins) [inline frontend/src/components/UserProfile.tsx:39]: These Centers make me a little uncomfortable. I think it'd probably be better to create a Flex that centers items in it so we don't have to keep repeating them. Not super important, just a thing that stuck out to me.

**Review by samdatkins — APPROVED** [review]
Nothing blocking, but a few changes you might care to look at

File: frontend/src/components/UserProfile.tsx (hunk 1 above)
Comment (jackconwayiv) [reply frontend/src/components/UserProfile.tsx:7]: My apologies. I copied from the useAsync example in SongbookList, but I failed to note that the async callback in that instance requires an argument.

File: frontend/src/components/UserProfile.tsx (hunk 3 above)
Comment (jackconwayiv) [reply frontend/src/components/UserProfile.tsx:39]: Thanks, I fixed this!

**Comment by jackconwayiv** [comment]
Avatar has been replaced by Image to include no-referrer.

File: frontend/src/components/UserProfile.tsx (hunk 2 above)
Comment (jackconwayiv) [reply frontend/src/components/UserProfile.tsx:23]: It is done.

**Review by samdatkins — APPROVED** [review]
Looks good to me!

**Pull request merged**